### DISCUSSION TITLE
v2 Discussion

### DISCUSSION THREAD

**Discussion by AathifMahir** [AathifMahir/MauiIcons#58] [General]
Hello Folks,

Since the inception of Maui icon Module, I was working on improving build by build but reach the point where whenever we need new features, we are in the position of rewriting around 70 to 80% of the core. Therefore I have decided to go hard on the problem and rearchitect the core to be much more modular and extensible. Therefore I'm creating this discussion bring in ideas and feedback on this issue.

**Comment by vhugogarcia** [comment]
Hello @AathifMahir ,

I would like to share the features I think will be awesome for version 2:

* Separated packages for multiple font icons, font icons files tend to be heavy on size, so if I just want to use one font icon on the app I would like the package to be as minimum as possible in size. 
* Font Weights, for example Material Symbols support weight, I think it would be super helpful to add the ability to set the weight for the icon. 
* Label support, add the ability to use the font icon on labels. 
* Remove icon custom view, I believe that by adding support to the native Maui controls is more than enough, we as developers just want to use the icon on what already exists. 

These are just my grain of salts

Regards,
Victor

**Reply by AathifMahir** [reply]
> Hello @AathifMahir ,
> 
> I would like to share the features I think will be awesome for version 2:
> 
> * Separated packages for multiple font icons, font icons files tend to be heavy on size, so if I just want to use one font icon on the app I would like the package to be as minimum as possible in size. 
> * Font Weights, for example Material Symbols support weight, I think it would be super helpful to add the ability to set the weight for the icon. 
> * Label support, add the ability to use the font icon on labels. 
> * Remove icon custom view, I believe that by adding support to the native Maui controls is more than enough, we as developers just want to use the icon on what already exists. 
> 
> These are just my grain of salts
> 
> Regards,
> Victor

Thanks man, that's very useful 👌
